plugins/parquet/parquet.cpp

```diff
@@ -6,511 +6,326 @@
-      arrays.push_back(new_arr);
-    else
-      arrays.push_back(table->column(i));
+auto parse_parquet(generator<chunk_ptr> input, operator_control_plane& ctrl)
+  -> generator<table_slice> {
+  auto parquet_chunk = chunk_ptr{};
+  for (auto&& chunk : drain_bytes(std::move(input))) {
+    if (not chunk) {
+      co_yield {};
+      continue;
+    }
+    TENZIR_ASSERT(not parquet_chunk);
+    parquet_chunk = std::move(chunk);
   }
-  auto new_table = arrow::Table::Make(target_schema, arrays, table->num_rows());
-  const auto delta = std::chrono::steady_clock::now() - start;
-  TENZIR_DEBUG("table schema aligned in {}[ns]",
-               data{std::chrono::duration_cast<duration>(delta)});
-  return new_table;
-}
-
-auto derive_import_time(const std::shared_ptr<arrow::Array>& time_col) {
-  return value_at(time_type{}, *time_col, time_col->length() - 1);
-}
-
-/// Extract event column from record batch and transform into new record batch.
-/// The record batch contains a message envelope with the actual event data
-/// alongside Tenzir-related meta data (currently limited to the import time).
-/// Message envelope is unwrapped and the metadata, attached to the to-level
-/// schema the input record batch is copied to the newly created record batch.
-std::shared_ptr<arrow::RecordBatch>
-unwrap_record_batch(const std::shared_ptr<arrow::RecordBatch>& rb) {
-  auto event_col = rb->GetColumnByName("event");
-  auto schema_metadata = rb->schema()->GetFieldByName("event")->metadata();
-  auto event_rb = arrow::RecordBatch::FromStructArray(event_col).ValueOrDie();
-  return event_rb->ReplaceSchemaMetadata(schema_metadata);
-}
-
-/// Create multiple table slices for a record batch, splitting at `max_slice_size`
-std::vector<table_slice>
-create_table_slices(const std::shared_ptr<arrow::RecordBatch>& rb,
-                    int64_t max_slice_size) {
-  auto final_rb = unwrap_record_batch(rb);
-  auto time_col = rb->GetColumnByName("import_time");
-  auto slices = std::vector<table_slice>{};
-  slices.reserve(rb->num_rows() / max_slice_size + 1);
-  auto schema = type::from_arrow(*final_rb->schema());
-  for (int64_t offset = 0; offset < rb->num_rows(); offset += max_slice_size) {
-    auto rb_sliced = final_rb->Slice(offset, max_slice_size);
-    auto& slice = slices.emplace_back(rb_sliced, schema);
-    slice.import_time(
-      derive_import_time(time_col->Slice(offset, max_slice_size)));
-    slice.offset(detail::narrow_cast<id>(offset));
+  auto input_file = as_arrow_file(std::move(parquet_chunk));
+  auto parquet_reader_properties
+    = ::parquet::ReaderProperties(arrow::default_memory_pool());
+  parquet_reader_properties.enable_buffered_stream();
+  std::unique_ptr<::parquet::arrow::FileReader> out_buffer;
+  auto arrow_reader_properties = ::parquet::ArrowReaderProperties();
+
+  // 2^16 is the largest batch size allowed in a table slice
+  arrow_reader_properties.set_batch_size(65536);
```

Comment:
```suggestion
  arrow_reader_properties.set_batch_size(defaults::import::table_slice_size);
```

(and remove the comment)